File: Code/app/src/main/java/com/bigfriedicecream/overcooked/observables/EventsDispatcher.java
package com.bigfriedicecream.overcooked.observables;

import java.util.HashMap;
import java.util.Observable;

public class EventsDispatcher extends Observable {

    private static EventsDispatcher instance;

    private EventsDispatcher() {}

    public static EventsDispatcher getInstance() {
        if (instance == null) {
            instance = new EventsDispatcher();
        }
        return instance;
    }

    public void dispatch(String name, String id) {
        HashMap<String, String> hashMap = new HashMap<>();
        hashMap.put("name", name);
        hashMap.put("id", id);
        setChanged();
        notifyObservers(hashMap);
    }
}

File: Code/app/src/main/java/com/twobrothers/overcooked/observables/EventsDispatcher.kt
package com.twobrothers.overcooked.observables

import java.util.HashMap
import java.util.Observable

class EventsDispatcher private constructor() : Observable() {

    private object Holder {
        val INSTANCE = EventsDispatcher()
    }

    companion object {
        val instance:EventsDispatcher by lazy { Holder.INSTANCE }
    }

    fun dispatch(name: String, id: String) {
        val hashMap = HashMap<String, String>()
        hashMap["name"] = name
        hashMap["id"] = id
        setChanged()
        notifyObservers(hashMap)
    }

}
